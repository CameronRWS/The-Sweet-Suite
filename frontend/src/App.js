import { BrowserRouter, Route, Switch, Redirect } from "react-router-dom";
import Login from "./pages/Login";
import GameSuite from "./pages/GameSuite";
import Checkers from "./pages/Checkers";
import Signup from "./pages/Signup";
import Settings from "./pages/Settings";
import React, {useState} from "react";

function App() {
  return (
    <div>
      <BrowserRouter>
        <Switch>
          <Route exact path="/">
            <Login></Login>
          </Route>
          <Route path="/gamesuite">
            <GameSuite></GameSuite>
          </Route>
          <Route path="/checkers">
            <Checkers></Checkers>
          </Route>
          <Route path="/signup">
            <Signup></Signup>
          </Route>
          <Route path="/settings">
            <Settings></Settings>
            </Route>
        </Switch>
      </BrowserRouter>
    </div>
  );
}

export default App;
