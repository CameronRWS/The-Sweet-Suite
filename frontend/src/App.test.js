import React from 'react';
import ReactDOM from 'react-dom';
import { render, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import App from './App';

// test('example test', () => {
//   render(<App />);
//   const paraElement = screen.getByText(/The Sweet Suite/i);
//   expect(paraElement).toBeInTheDocument();
// });

let container;

beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    document.body.removeChild(container);
    container = null;
});

it('states your login credentials', () => {
  act(() => {
      ReactDOM.render(<App/>, container);
  });

  const resetButtonText = container.querySelector('div');
  expect(resetButtonText.textContent).toContain('Login');
});