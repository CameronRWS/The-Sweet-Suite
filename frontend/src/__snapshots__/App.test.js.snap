// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h1>
        The Sweet Suite
      </h1>
      <h2>
        Login
      </h2>
      <form>
        <div>
          <label>
            User Name
          </label>
          <input
            data-test="user"
            type="text"
            value=""
          />
        </div>
        <div>
          <label>
            Password
          </label>
          <input
            data-test="pass"
            type="password"
            value=""
          />
        </div>
        <button
          type="submit"
        >
          Login
        </button>
      </form>
      <button>
        Signup
      </button>
    </div>
  </div>
</DocumentFragment>
`;
